src/slack/conversation.go

```diff
@@ -60,3 +62,15 @@ func (c *conversation) Get(key string) ([]string, bool) {
 	value, ok := c.data[key]
 	return value, ok
 }
+
+// ClearConversation delete current conversation history
+func (c *conversation) ClearConversation(key string, convo *conversation) bool {
+	// Key exists in the map delete it from map and return true
```

Comment:
there is a commonality between the other methods used that is important to understand as it relates to concurrent operations on a map. 

```
c.Lock()
defer c.Unlock()
```
Are present in the other operations because they prevent other operations from happening on the map concurrently. 

It is possible that 2 separate goroutines are operating on this map at the same time. What if the other goroutine tries to add data to this key at the same time as this method is trying to delete data?

Concurrency, and thinking concurrently, is a challenging skill to develop. I hope this is helpful. If you need more explanation, hit me up!

Reply:
Will do man thanks, I think I'm tracking but I definitely wouldn't mind a quick chat just to learn. I'm gonna dive into all the comments tomorrow and see if I can wrap this one up.